azure/scope/machine.go

```diff
@@ -191,16 +194,21 @@ func (m *MachineScope) PublicIPSpecs() []azure.PublicIPSpec {
 }
 
 // InboundNatSpecs returns the inbound NAT specs.
-func (m *MachineScope) InboundNatSpecs() []azure.InboundNatSpec {
+func (m *MachineScope) InboundNatSpecs(existingRules []network.InboundNatRule) []azure.ResourceSpecGetter {
 	if m.Role() == infrav1.ControlPlane {
-		return []azure.InboundNatSpec{
-			{
-				Name:             m.Name(),
-				LoadBalancerName: m.APIServerLBName(),
+		ipConfig := m.APIServerLB().FrontendIPs[0].Name
```

Comment:
// TODO: need to figure out how this should be handles when the LB has more than one frontend IP

Reply:
It looks like we have a check in `azurecluster_validation.go` to ensure there's exactly one frontend IP, so it looks like we'd only need to handle that if our spec invariant changes.

```
// There should only be one IP config.
if len(lb.FrontendIPs) != 1 || pointer.Int32Deref(lb.FrontendIPsCount, 1) != 1 {
	allErrs = append(allErrs, field.Invalid(fldPath.Child("frontendIPConfigs"), lb.FrontendIPs,
		"API Server Load balancer should have 1 Frontend IP"))
} 
```